### PR TITLE
Changed: new hooks will use givewp as a prefix

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

As we continue to improve our codebase, new actions and filters are being created.  To prevent conflict with legacy hooks, the new prefix will be `givewp_` instead of `give_`.  This PR updates the recently added hooks in our model repositories to use this naming convention.

## Affects

- Donor, Donation, and Subscription repository actions.

## Visuals

Screen Shot 2022-05-20 at 2 53 04 PM

## Testing Instructions

These hooks were just released to there is not much to test.  You can use the donation form to create some donations and make sure there's no issues.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@JasonTheAdams this glorious PR is ready for review.

File: src/Donors/Repositories/DonorRepository.php [hunk 1]
```diff
@@ -33,11 +34,8 @@ class DonorRepository
      * Query Donor By ID
      *
      * @since 2.19.6
-     *
-     * @param int $donorId
-     * @return ModelQueryBuilder
```
Comment (JasonTheAdams) [inline src/Donors/Repositories/DonorRepository.php:38]: I think we want to leave this with the generic: `@return ModelQueryBuilder<Donor>`. Same for the other usages.

File: src/Subscriptions/Repositories/SubscriptionRepository.php [hunk 2]
```diff
@@ -204,28 +189,23 @@ public function update(Subscription $subscription)
 
         DB::query('COMMIT');
 
-        $subscriptionId = DB::last_insert_id();
-
-        $subscription = $this->getById($subscriptionId);
-
-        Hooks::doAction('give_subscription_updated', $subscription);
-
-        return $subscription;
```
Comment (JasonTheAdams) [inline src/Subscriptions/Repositories/SubscriptionRepository.php:213]: Do we need to update the `save()` method in the model as well?

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Love it! Just a couple small things.

File: src/Subscriptions/Repositories/SubscriptionRepository.php (hunk 2 above)
Comment (jonwaldstein) [reply src/Subscriptions/Repositories/SubscriptionRepository.php:213]: Nope! that was already done strangely enough.

File: src/Donors/Repositories/DonorRepository.php (hunk 1 above)
Comment (jonwaldstein) [reply src/Donors/Repositories/DonorRepository.php:38]: Ah yeah good point, I wasn't sure if that would still work but probably not

Comment (jonwaldstein) [reply src/Donors/Repositories/DonorRepository.php:38]: I added back the generics and did an overhaul of docblock cleanups for the models

File: src/Donors/Models/Donor.php [hunk 3]
```diff
@@ -196,7 +195,7 @@ public function totalAmountDonated(): int
      *
      * @return ModelQueryBuilder<Donor>
      */
-    public static function query(): ModelQueryBuilder
```
Comment (JasonTheAdams) [inline src/Donors/Models/Donor.php:199]: Missed one!

**Review by JasonTheAdams — APPROVED** [review]
Great work! Just one docblock missed, but otherwise good to go!

**Pull request merged**